internal/controller/pkg/revision/watch.go

```diff
@@ -89,19 +90,45 @@ func EnqueuePackageRevisionsForImageConfig(kube client.Client, l v1.PackageRevis
 		rl := l.DeepCopyObject().(v1.PackageRevisionList) //nolint:forcetypeassert // Guaranteed to be PackageRevisionList.
 		if err := kube.List(ctx, rl); err != nil {
 			// Nothing we can do, except logging, if we can't list FunctionRevisions.
-			log.Debug("Cannot list function revisions while attempting to enqueue from ImageConfig", "error", err)
+			log.Debug("Cannot list package revisions while attempting to enqueue from ImageConfig", "error", err)
 			return nil
 		}
 
 		var matches []reconcile.Request
 		for _, rev := range rl.GetRevisions() {
 			for _, m := range ic.Spec.MatchImages {
 				if strings.HasPrefix(rev.GetSource(), m.Prefix) {
-					log.Debug("Enqueuing %T for image config", "functionRevision", rev, rev.GetName(), "imageConfig", ic.GetName())
+					log.Debug("Enqueuing for image config",
+						"revision-type", fmt.Sprintf("%T", rev),
+						"revision-name", rev.GetName(),
+						"image-config-name", ic.GetName())
 					matches = append(matches, reconcile.Request{NamespacedName: types.NamespacedName{Name: rev.GetName()}})
 				}
 			}
 		}
 		return matches
 	})
 }
+
+// EnqueuePackageRevisionsForLock enqueues a reconcile for all package
+// revisions when the Lock changes.
+func EnqueuePackageRevisionsForLock(kube client.Client, l v1.PackageRevisionList, log logging.Logger) handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o client.Object) []reconcile.Request {
+		if _, ok := o.(*v1beta1.Lock); !ok {
+			return nil
+		}
+
+		rl := l.DeepCopyObject().(v1.PackageRevisionList) //nolint:forcetypeassert // Guaranteed to be PackageRevisionList.
+		if err := kube.List(ctx, rl); err != nil {
+			// Nothing we can do, except logging, if we can't list FunctionRevisions.
```

Comment:
nit: should this comment be `PackageRevisions` instead of `FunctionRevisions`?

Reply:
Yeah, or "package revisions" since `PackageRevision` isn't a concrete type. Will update before merging.